Sources/Core/AWSClientRuntime/UserAgent/ExecutionEnvMetadata.swift

```diff
@@ -18,6 +18,9 @@ extension ExecutionEnvMetadata: CustomStringConvertible {
     }
 
     static func detectExecEnv() -> ExecutionEnvMetadata? {
+        #if targetEnvironment(simulator)
+        return ExecutionEnvMetadata(name: "simulator")
```

Comment:
From the user-agent SEP:
> Execution environment name can be retrieved from the environment variable AWS_EXECUTION_ENV as defined in this [SEP](https://code.amazon.com/packages/AwsDrSeps/blobs/ca26ae803530fb10429a218c1c0a3e25ee07cceb/--/seps/accepted/shared/user_agent.md). The SDK MUST include this in the header if present.

I don't think simulator is the intended use of this UA field, and I'm afraid we will mess up data collection if we put other values into this field.

Reply:
That's fair, I think what I could do is append this as additional metadata to OS metadata then.